src/proc/motion-transform.cpp

```diff
@@ -12,34 +12,53 @@
 
 namespace librealsense
 {
-    template<rs2_format FORMAT> void copy_hid_axes( uint8_t * const dest[], const uint8_t * source, double factor, bool is_mipi)
+    template<rs2_format FORMAT> void copy_hid_axes( uint8_t * const dest[], const uint8_t * source, double factor, bool high_accuracy, bool is_mipi)
     {
         using namespace librealsense;
 
         auto res = float3();
-        // D457 dev
+
         if (is_mipi)
         {
             auto hid = (hid_mipi_data*)(source);
-            res = float3{ float(hid->x), float(hid->y), float(hid->z) } *float(factor);
+
+            if( ! high_accuracy )
+            {
+                //since D400 FW version 5.16 the hid report struct changed to 32 bit for each paramater.
+                //To support older FW versions we convert the data to int16_t before casting to float.
```

Comment:
`as we only get valid data at the lower 16  bits `

Reply:
Done